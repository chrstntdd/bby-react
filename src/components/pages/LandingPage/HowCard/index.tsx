import * as React from 'react';
import './how-card.scss';

const Card = props => (
  <div className="how-card">
    <img src={props.img} alt={props.name + ' icon'} />
    <h3>{props.name}</h3>
    <h5>{props.desc}</h5>
  </div>
);

export default Card;
