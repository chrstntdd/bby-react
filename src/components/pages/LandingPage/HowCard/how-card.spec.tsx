import * as React from 'react';
import { shallow } from 'enzyme';

import HowCard from './';

describe('<HowCard/>', () => {
  const props = {
    cardData: [
      {
        name: 'simple',
        img: 'fakeImgUrl.jpg',
        desc: '...but powerful'
      },
      {
        name: 'reliable',
        img: 'fakeImgUrl.jpg',
        desc: 'There when you need it'
      },
      {
        name: 'safe',
        img: 'fakeImgUrl.jpg',
        desc: 'Without compromises'
      },
      {
        name: 'fast',
        img: 'fakeImgUrl.jpg',
        desc: 'Designed with speed in mind'
      }
    ]
  };
  it('should render without crashing', () => {
    const wrapper = shallow(<HowCard {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should have an image, h3, and h5', () => {
    const wrapper = shallow(<HowCard {...props} />);
    const expectedTags = ['img', 'h3', 'h5'];
    const renderedTags = wrapper.children().nodes.map(tag => tag.type);
    expect(expectedTags).toEqual(renderedTags);
  });
});
