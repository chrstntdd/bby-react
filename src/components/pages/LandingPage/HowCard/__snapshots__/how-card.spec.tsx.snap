// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HowCard/> should render without crashing 1`] = `
<div
  className="how-card"
>
  <img
    alt="undefined icon"
  />
  <h3 />
  <h5 />
</div>
`;
